### PR TITLE
Protect Status: Provide unified threats list

### PR DESCRIPTION
**Pull request by nateweller**
## Proposed changes:

* Adds a `threats: Threat[]` property to the Protect Status.
* Deprecates properties such as `plugins`, `themes`, `core`, and `files` - the threats provided by these properties can now be accessed from the single `threats` property.
* Deprecates additional computed properties such as `num_threats`, `num_plugin_threats`, etc – these values can be computed by consumers using the `threats` property.
* **Only uses `@deprecated` PHPDoc blocks, does not include use of core `_deprecated_*_()` functions, to prevent PHP warnings from being triggered by an intentional deprecation process** (is this ridiculous? my intention is to avoid support tickets from users with `WP_DEBUG` enabled)
* Updates tests.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1730827086851209-slack-C029WFNV69M

peb6dq-31T-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Proof changes, ensure tests are passing.
* While viewing the Protect admin screen, check the contents of `window.jetpackProtectInitialState` to validate the presence of the new `threats` property.
* Test for regression:
  * Jetpack Protect: Test scanning and viewing threat results on both a free and paid plan.
  * My Jetpack: Test viewing the Protect product card in inactive/active states.

### CODE REVIEW

File: projects/packages/protect-models/src/class-extension-model.php
Comment (nateweller) [inline projects/packages/protect-models/src/class-extension-model.php:1]: * Deprecate the `Extension_Model::threats` property, as we will now use the inverse `Threat_Model::extension` relationship.
* Deprecate the `set_threats` method as it now applies to a deprecated property.

File: projects/packages/protect-models/src/class-history-model.php
Comment (nateweller) [inline projects/packages/protect-models/src/class-history-model.php:1]: * Add `History_Model::$threats`, the unified threats list which replaces the separate `plugins`, `themes`, `core`, `files`, and `database` properties.

File: projects/packages/protect-models/src/class-status-model.php
Comment (nateweller) [inline projects/packages/protect-models/src/class-status-model.php:1]: * Add Status_Model::$threats, the unified threats list which replaces the separate plugins, themes, core, files, and database properties.
* Also deprecate the computed count properties for the separate plugin/theme/etc categories.

File: projects/packages/protect-models/src/class-threat-model.php
Comment (nateweller) [inline projects/packages/protect-models/src/class-threat-model.php:1]: * Add `Threat_Model::$extension` to replace the now deprecated, inverse `Extension_Model::$threats` property.

**Review by dkmyta — APPROVED** [review]
Looks, tests, and works great!

>Only uses @deprecated PHPDoc blocks

This is the only thing that I cannot confidently address but outside of this, IMO this is ready to ship. If you are entirely satisfied that this is the correct approach let's press forward, but if not, perhaps its worth wrangling in someone from Jetpack development with experience in the matter for a second or third opinion.

**Pull request merged**